python/eos/tasks.py

```diff
@@ -412,7 +412,17 @@ def predict_observables(analysis_file:str, posterior:str, prediction:str, base_d
 
     parameters = [_parameters[p['name']] for p in data.varied_parameters]
     observable_samples = []
-    for i, sample in enumerate(progressbar(data.samples[begin:end])):
+
+    if end == -1:
```

Comment:
All this can be avoided by using ``None`` as the default. Sorry for this, I am pretty sure that I scewed it up in the first place.